ocs_ci/ocs/resources/cloud_manager.py

```diff
@@ -127,6 +127,28 @@ def get_all_uls_names(self):
     def verify_uls_exists(self, uls_name):
         pass
 
+    def verify_uls_state(self, uls_name, is_available):
+        check_type = "Delete"
+        if is_available:
+            check_type = "Create"
+        sample = TimeoutSampler(
+            timeout=180, sleep=15, func=self.verify_uls_exists, uls_name=uls_name
+        )
+        if sample.wait_for_func_status(result=is_available):
+            logger.info(
+                f"Underlying Storage {uls_name} {check_type.lower()}d successfully."
+            )
+        else:
+            if is_available:
+                raise ResourceInUnexpectedState(
+                    f"{check_type}ion of Underlying Storage {uls_name} timed out. "
+                    f"Unable to {check_type.lower()} {uls_name}"
+                )
+            else:
+                logger.warning(
```

Comment:
in case the resource still exists after its deletion, shouldn't we raise an exception?

Reply:
We used to; But we removed it because ULS deletion isn't a synchronous task, and it sometimes may take a few hours. Thus, it'd lead to test failures, even though the ULSs were deleted after a few hours

Reply:
will it be possible to send a notification when a resource deletion fails? We don't want to continue paying for such leftovers